install-dependencies.sh

```diff
@@ -80,73 +81,140 @@ if [ -z "$DISTRO" ]; then
    echo "Installing dependencies for FreeBSD..."
    pkg install git asio cmake gettext glew png sdl2_image sdl2_mixer sdl2_net sdl2_ttf
+
 elif [ "$DISTRO" == "openbsd" ]; then
    echo "Installing dependencies for OpenBSD..."
-   pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
+   doas pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
     sdl2-image sdl2-mixer sdl2-net sdl2-ttf
-   echo "Asio is not packaged for OpenBSD, please install it from source:"
-   echo " - Download the latest stable version from"
-   echo "      https://sourceforge.net/projects/asio/files/asio/"
-   echo " - Unpack in your development directory"
-   echo " - Run './configure --prefix=/usr && make install-data' in the unpacked asio"
-   echo "   directory (or just copy recursively 'asio.hpp' and the 'asio' subdirectory"
-   echo "   from 'include' to '/usr/include')"
+   if ! [ -f /usr/include/asio.hpp -o -f ./auto_dependencies/asio/asio.hpp ]
+   then
+      asio_not_packaged "OpenBSD" "doas"
+      exit 1
+   fi
+
 elif [ "$DISTRO" == "msys32" ]; then
    echo "Installing dependencies for 32-bit Windows..."
    pacman -S pacman -S mingw-w64-i686-toolchain git mingw-w64-i686-cmake \
     mingw-w64-i686-ninja mingw-w64-i686-asio mingw-w64-i686-SDL2_ttf \
     mingw-w64-i686-SDL2_mixer mingw-w64-i686-SDL2_image \
     mingw-w64-i686-glbinding mingw-w64-i686-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "msys64" ]; then
    echo "Installing dependencies for 64-bit Windows..."
    pacman -S mingw-w64-x86_64-toolchain git mingw-w64-x86_64-cmake \
     mingw-w64-x86_64-ninja mingw-w64-x86_64-asio mingw-w64-x86_64-SDL2_ttf \
     mingw-w64-x86_64-SDL2_mixer mingw-w64-x86_64-SDL2_image \
     mingw-w64-x86_64-glbinding mingw-w64-x86_64-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "homebrew" ]; then
    echo "Installing dependencies for Mac Homebrew..."
    brew install asio git cmake doxygen gettext glew graphviz icu4c jpeg \
     libogg libpng libvorbis ninja python sdl2 sdl2_image sdl2_mixer sdl2_ttf zlib
+
+elif [ "$DISTRO" == "solus" ]; then
+   echo "Installing dependencies for Solus..."
+   sudo eopkg install -c system.devel
+   sudo eopkg install git gettext glew-devel libicu-devel libpng-devel sdl2-devel \
+    sdl2-image-devel sdl2-mixer-devel sdl2-ttf-devel python
+
+   if ! [ -f /usr/include/asio.hpp -o -f ./auto_dependencies/asio/asio.hpp ]
+   then
+      echo "Installing Asio from source..."
+      rm -r ./auto_dependencies/asio >/dev/null 2>/dev/null || true
+      mkdir -p ./auto_dependencies/asio
+      cd ./auto_dependencies/asio
+
+      if which wget > /dev/null
+      then
+         REMOTE="wget -O"
+      elif which curl > /dev/null
+      then
+         REMOTE="curl -o"
+      else
+         asio_not_packaged "Solus" "sudo"
+         exit 1
+      fi
+
+      $REMOTE sources https://sourceforge.net/projects/asio/files/asio/1.20.0%20%28Stable%29/asio-1.20.0.tar.bz2/download
```

Comment:
```suggestion
      $REMOTE sources https://downloads.sourceforge.net/project/asio/asio/1.20.0%20%28Stable%29/asio-1.20.0.tar.bz2
```
This uses one less redirect - only to the closest mirror.